server/routes/charts.js

```diff
@@ -1,30 +1,71 @@
 import { Router } from 'express';
+import { collections } from '../utils/mongoCollections'
 import chartsListPage from '../templates/Chart.template'
 import newChartPage from '../templates/NewChart.template'
 import ensureAuthenticated from '../utils/passport/ensure-authenticated';
 
 const router = Router();
 
 router.route('/charts')
-  .get(ensureAuthenticated, async (_, res) => {
+  .get(ensureAuthenticated, async (req, res) => {
     try {
       return res.status(200).send(chartsListPage())
-    } catch (err) {
-      console.error(err.message)
+    } catch (error) {
+      console.error(error.message)
 
       return res.status(500).send({ message: err.message })
     }
   });
   
 router.route('/charts/new')
-  .get(ensureAuthenticated, async (_, res) => {
+  .get(ensureAuthenticated, async (req, res) => {
     try {
-      return res.status(200).send(newChartPage())
+      const { display_name, role, icon } = req.session;
+      const user = {
+        uid: req.user,
+        name: display_name,
+        role, 
+        icon,
+      };
+
+      return res.status(200).send(newChartPage(user))
     } catch (error) {
-      console.error(err.message)
+      console.error(error.message)
 
       return res.status(500).send({ message: err.message })
     }
 })
 
+router.route('/api/v1/chart-create')
+  .post(ensureAuthenticated, async (req, res) => {
+    try {
+      const additionalVariables = {}
+      const { fields, title, variable, assessment, owner } = req.body
+      
+      if (fields.length) {
+        for (let index = 0; index < fields.length; index++) {
+          const field = fields[index];
+          additionalVariables[field.label] = field.value
+        }
+      }
+
+      const dataDb = req.app.locals.dataDb
+      const { result } = await dataDb
+        .collection(collections.charts)
+        .insertOne({
+          title, 
+          variable, 
+          assessment, 
+          owner, 
+          ...additionalVariables 
```

Comment:
should we store these variables in a key, maybe something like `fieldLabelValueMap`? Maybe we even use an array of label/value objects (`req.body.fields`) for this? Also, since we know we are going to need to support dynamic values/labels should we have another variable like `useContinuousVariables` set to `false`?

Reply:
[fd611f7](https://github.com/AMP-SCZ/dpdash/pull/84/commits/fd611f7e8aaef083e1cbd59a407f40a134bd1863)